### ISSUE TITLE
Model Monitoring feature integration with Pycaret

### ISSUE THREAD

**Issue by aniketkumar430** [pycaret/pycaret#1448]
Pycaret has nice set of features. 
Addition of model monitoring features can be a big value addition to the library.

To do this I came across one open source library which can make this possible. There can be alternate possible libraries . But I am just sharing one of them below :
https://github.com/evidentlyai/evidently?

This can help in monitoring the  Models and the data in production.

**Comment by pycaret** [comment]
@aniketkumar430 Will you be interested in working on this? Let me know.

**Comment by wkuopt** [comment]
I think this would be a good integration to detect data and model issues in production. @pycaret Tensorflow Extended also provides similar functionalities: https://www.tensorflow.org/tfx/guide/tfdv. It's very comprehensive but of course heavier :)

**Comment by Yard1** [comment]
@wkuopt If possible, we will want to introduce an abstraction layer to enable different monitoring backends. I have started work on something similar for logging in https://github.com/pycaret/pycaret/pull/1276 - @AyushExel is working on expanding it a little and adding Weights & Biases integration.

**Comment by wkuopt** [comment]
Thanks @Yard1. Yes a layer to enable different backend definitely sounds good. I feel there is a bit difference between these two groups:
Group 1: weights & bias, MLflow
Group 2: evidently, TFX

Group 1 seems mostly focused on experiment tracking and metric reporting. Group 2 seems more about detecting data drift and model anomaly, fairness concerns. Maybe we can have different layers to deal with both?

**Comment by Yard1** [comment]
Yes, having two separate layers will be fine

**Comment by moezali1** [comment]
@Yard1 Do you have any updates from the W&B guy? The last time I heard he went on a vacation and then the team changed. Would be good to follow up with them. Would be good to introduce this integration in `2.3.5` and so it can be stable by `3.0`.

**Comment by huanvo88** [comment]
Any update on this ticket? MLflow has been integrated into pycaret. From what I've seen WANDB is more superior than MLflow, with Sweep and auto-generated reports, but not sure how easy it is to integrate WANDB to an open source, since I think you need to sign up and run things on their platform.

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
